front/src/modules/timesStops/helpers/utils.ts

```diff
@@ -56,21 +70,21 @@ export const formatSuggestedViasToRowVias = (
     const { arrival, onStopSignal, stopFor, theoreticalMargin } = objectToUse || {};
 
     const isMarginValid = theoreticalMargin ? marginRegExValidation.test(theoreticalMargin) : true;
-    let departure: string | undefined;
-    if (stopFor) {
-      if (i === 0) {
-        departure = startTime
-          ? secToHoursString(datetime2sec(new Date(startTime)) + Number(stopFor), true)
-          : undefined;
-      } else if (arrival) {
-        departure = secToHoursString(time2sec(arrival) + Number(stopFor), true);
-      }
-    }
+    const roundedArrivalTime = i === 0 ? 'PT0S' : arrival;
+    const arrivalInSeconds = roundedArrivalTime ? time2sec(roundedArrivalTime) : null;
+
+    const formattedArrival = calculateStepTimeAndDays(startTime, roundedArrivalTime);
+
+    const departureTime =
+      stopFor && arrivalInSeconds
+        ? secToHoursString(arrivalInSeconds + Number(stopFor), true)
```

Comment:
Instead of just passing true maybe it's clearer like that (and we add a `const WITH_SECONDS = true;` at the top of the file)

```suggestion
        ? secToHoursString(arrivalInSeconds + Number(stopFor), WITH_SECONDS)
```

Reply:
I’m a bit on the fence regarding that. I don’t think we want to create a variable for every single function that takes a boolean flag. 

It’s just a mouse hover away to get the info.

![Screenshot_20240913_164345](https://github.com/user-attachments/assets/c97b982a-bc87-43f9-bbe0-2542691be057)

@clarani @emersion what do you think?

Reply:
In general I'd prefer options objects e.g.

```ts
secToHoursString(arrivalInSeconds + Number(stopFor), { withSeconds: true })
```

since it's an idiomatic design pattern for JS APIs and doesn't require readers to have a full IDE set up (useful for reading on git(1) or GitHub, and for weird minimalists like me).

Reply:
I think it would be better to have a function secToHhmmss which calls secToHoursString with true, what do you think ?
Also we could rename the function secToHoursString into secToHhmmString, but maybe in another PR

Reply:
I also agree with @emersion ;)

Reply:
TBQH, I think the utils date API is due for a big overhaul to improve/unify all exposed functions. It's quite inconsistent at the moment.

Reply:
I also like @emersion suggestion

Reply:
going for @emersion’s suggestion as we seem to all agree :)